### DISCUSSION TITLE
Qt and simdjson: best way to read strings?

### DISCUSSION THREAD

**Discussion by vadi2** [simdjson/simdjson#1442] [Q&A] [answered]
Trying to read data into a QString in an exception-free manner, one can write this:

```cpp
    std::string_view temp;
    error = simd_doc["defaultAreaName"].get<std::string_view>(temp);
    mDefaultAreaName = QString::fromUtf8(temp.data(), temp.size());
```

Is this the most efficient way to go?

**Comment by lemire** [comment, marked as answer]
1. You should not need the template... please follow our examples... `error = simd_doc["defaultAreaName"].get(temp)`.
2. Your code does not do any error checking. It would be safer to do:
```
  std::string_view temp = simd_doc["defaultAreaName"];
  mDefaultAreaName = QString::fromUtf8(temp.data(), temp.size());
```
3. As I stated in a related issue, you should benchmark the `QString::fromUtf8` function. You may find that it is surprisingly expensive.
4. You can, of course, write your own convenience functions.

**Reply by jkeiser** [reply]
I can't see how to make it simpler without either exceptions or ignoring the error entirely ... it seems like that's just one of the costs of running without exceptions. Super interested to hear a way, though! Maybe there's a subset of errors you could ignore and pass a desired default value to get_string()? Still going to be unrecoverable errors like eof, though ...

**Reply by vadi2** [reply]
I don't think I have a choice regarding `fromUtf8()`, at the of the day I need a [QString](https://doc.qt.io/qt-5/qstring.html), not a `std::string_view`, as nice as the latter is.

**Reply by jkeiser** [reply]
Oh, I was talking less about the qstring, conversion is sort of a requirement. Supporting custom converters like nlohmann_json might cut out one step though, it's worth looking into.

**Reply by vadi2** [reply]
I think that would be great.

**Comment by jkeiser** [comment]
Note that simdjson doesn't presently give you any guarantees that your variable will get a value if there is an error: if you are going with no exceptions, which is totally supported, you should probably wrap that call to at with an if statement.

**Reply by vadi2** [reply]
Right, I will.
